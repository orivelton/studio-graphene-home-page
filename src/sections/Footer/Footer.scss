@import '../../assets/scss/helper.scss';

.footer {
  background: $night-rider;
  padding: 63px 0;
  display: flex;
  flex-direction: column;
  align-items: center;
  &__box {
    display: flex;
    flex-direction: row;
    width: 1072px;
    justify-content: space-between;
  }

  &__links {
    display: flex;
    flex-direction: row;
  }
}

.pages {
  display: flex;
  flex-direction: column;
  width: 135px;
  &__title {
    color: $cinderella;
    font-size: 14px;
    margin-bottom: 14px;;
  }

  &__link {
    font-family: 'Gill Sans';
    text-decoration: none;
    color: $grey;
    font-size: 13px;
    margin-bottom: 15px;
  }
}

.copyright {
  font-family: 'Gill Sans';
  text-decoration: none;
  color: $grey;
  font-size: 13px;
  margin-top: 33px;
  letter-spacing: 0.13px;
}
