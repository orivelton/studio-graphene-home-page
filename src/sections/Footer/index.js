import React from 'react';
import './Footer.scss';
import Title from '../../components/Title';
import { linksFooter } from '../../mock/mock-data';

const Footer = () => {

  const buildItemFooter = (links) => links.map((item, index) =>
    <a key={index} className="pages__link" href="/" title={`Go to ${item}`}>
      {item}
    </a>
  );

  return (
    <footer className="footer">
      <div className="footer__box">
        <div className="footer__links">
          {
            linksFooter.map((item, index) => {
              const { title, links } = item;

              return (
                <div className="pages" key={index}>
                  <p className="pages__title uppercase">{`${title} ${++index}`}</p>
                  {buildItemFooter(links)}
                </div>
              )
            })
          }

        </div>
        <Title textTop={'This is the'} text={'Logo'} colorSecondary />
      </div>

      <div className="footer__box">
        <p className="copyright">2017. Company. Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor .</p>
        <ul className="social">
          <li className="social__item">
            <a href="/" className="social__icon icon iconfacebook-icon" title="Go to Facebook">
              <span className="social__name">Facebook</span>
            </a>
          </li>
          <li className="social__item">
            <a href="/" className="social__icon icon icontwitter-icon" title="Go to twitter">
              <span className="social__name">twitter</span>
            </a>
          </li>
          <li className="social__item">
            <a href="/" className="social__icon icon iconinstagram-icon" title="Go to instagram">
              <span className="social__name">instagram</span>
            </a>
          </li>
          <li className="social__item">
            <a href="/" className="social__icon icon iconLinkedin-icon" title="Go to Linkedin">
              <span className="social__name">Linkedin</span>
            </a>
          </li>
        </ul>
      </div>
    </footer>
  );
};

export default Footer;