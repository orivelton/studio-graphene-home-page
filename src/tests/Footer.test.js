import React from 'react';
import renderer from 'react-test-renderer';
import Footer from '../sections/Footer';

test('renders correctly Footer', () => {
  const tree = renderer.create(<Footer />).toJSON();
  expect(tree).toMatchSnapshot();
});
