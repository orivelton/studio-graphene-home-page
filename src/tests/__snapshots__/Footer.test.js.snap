// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly Footer 1`] = `
<footer
  className="footer"
>
  <div
    className="footer__box"
  >
    <div
      className="footer__links"
    >
      <div
        className="pages"
      >
        <p
          className="pages__title uppercase"
        >
          Section 1
        </p>
        <a
          className="pages__link"
          href="/"
          title="Go to Subsection"
        >
          Subsection
        </a>
        <a
          className="pages__link"
          href="/"
          title="Go to Subsection"
        >
          Subsection
        </a>
        <a
          className="pages__link"
          href="/"
          title="Go to Subsection"
        >
          Subsection
        </a>
        <a
          className="pages__link"
          href="/"
          title="Go to Subsection"
        >
          Subsection
        </a>
      </div>
      <div
        className="pages"
      >
        <p
          className="pages__title uppercase"
        >
          Section 2
        </p>
        <a
          className="pages__link"
          href="/"
          title="Go to Subsection"
        >
          Subsection
        </a>
        <a
          className="pages__link"
          href="/"
          title="Go to Subsection"
        >
          Subsection
        </a>
        <a
          className="pages__link"
          href="/"
          title="Go to Subsection"
        >
          Subsection
        </a>
        <a
          className="pages__link"
          href="/"
          title="Go to Subsection"
        >
          Subsection
        </a>
      </div>
      <div
        className="pages"
      >
        <p
          className="pages__title uppercase"
        >
          Section 3
        </p>
        <a
          className="pages__link"
          href="/"
          title="Go to Subsection"
        >
          Subsection
        </a>
        <a
          className="pages__link"
          href="/"
          title="Go to Subsection"
        >
          Subsection
        </a>
        <a
          className="pages__link"
          href="/"
          title="Go to Subsection"
        >
          Subsection
        </a>
        <a
          className="pages__link"
          href="/"
          title="Go to Subsection"
        >
          Subsection
        </a>
      </div>
      <div
        className="pages"
      >
        <p
          className="pages__title uppercase"
        >
          Section 4
        </p>
        <a
          className="pages__link"
          href="/"
          title="Go to Subsection"
        >
          Subsection
        </a>
        <a
          className="pages__link"
          href="/"
          title="Go to Subsection"
        >
          Subsection
        </a>
        <a
          className="pages__link"
          href="/"
          title="Go to Subsection"
        >
          Subsection
        </a>
        <a
          className="pages__link"
          href="/"
          title="Go to Subsection"
        >
          Subsection
        </a>
      </div>
      <div
        className="pages"
      >
        <p
          className="pages__title uppercase"
        >
          Section 5
        </p>
        <a
          className="pages__link"
          href="/"
          title="Go to Subsection"
        >
          Subsection
        </a>
        <a
          className="pages__link"
          href="/"
          title="Go to Subsection"
        >
          Subsection
        </a>
        <a
          className="pages__link"
          href="/"
          title="Go to Subsection"
        >
          Subsection
        </a>
        <a
          className="pages__link"
          href="/"
          title="Go to Subsection"
        >
          Subsection
        </a>
      </div>
    </div>
    <div>
      <div
        className="title title--secondary uppercase"
      >
        <h3
          className="title__top"
        >
          THIS IS THE
          <span
            className="title__text"
          >
            Logo
          </span>
        </h3>
      </div>
    </div>
  </div>
  <div
    className="footer__box"
  >
    <p
      className="copyright"
    >
      2017. Company. Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor .
    </p>
    <ul
      className="social"
    >
      <li
        className="social__item"
      >
        <a
          className="social__icon icon iconfacebook-icon"
          href="/"
          title="Go to Facebook"
        >
          <span
            className="social__name"
          >
            Facebook
          </span>
        </a>
      </li>
      <li
        className="social__item"
      >
        <a
          className="social__icon icon icontwitter-icon"
          href="/"
          title="Go to twitter"
        >
          <span
            className="social__name"
          >
            twitter
          </span>
        </a>
      </li>
      <li
        className="social__item"
      >
        <a
          className="social__icon icon iconinstagram-icon"
          href="/"
          title="Go to instagram"
        >
          <span
            className="social__name"
          >
            instagram
          </span>
        </a>
      </li>
      <li
        className="social__item"
      >
        <a
          className="social__icon icon iconLinkedin-icon"
          href="/"
          title="Go to Linkedin"
        >
          <span
            className="social__name"
          >
            Linkedin
          </span>
        </a>
      </li>
    </ul>
  </div>
</footer>
`;
